integration_tests/tests/sts.rs

```diff
@@ -46,4 +49,16 @@ fn main() {
         Err(e) => panic!("sts credentials provider error: {:?}", e),
         Ok(r) => println!("sts credentials provider result: {:?}", r)
     }
+
+    // check that it's possible to create a new ec2 client with sts
+    let provider = StsAssumeRoleSessionCredentialsProvider::new(
+        sts,
```

Comment:
I think this line is causing the error: `sts` is moved on line 46: 

```rust
let sts_creds_provider = StsSessionCredentialsProvider::new(sts, None, None);
```

Should this new part of the test get a new `StsClient`?